crates/move-prover-boogie-backend/src/boogie_backend/bytecode_translator.rs

```diff
@@ -1804,6 +1849,17 @@ impl<'env> FunctionTranslator<'env> {
             emitln!(writer, "$t{} := _$t{};", idx, idx);
         }
 
+        // Initialize ghost variables
+        if self.style == FunctionTranslationStyle::Default
+            || self.style == FunctionTranslationStyle::Opaque
+        {
+            let ghost_vars = self.get_ghost_vars();
+            for type_inst in ghost_vars {
+                let var_name = boogie_spec_global_var_name(self.parent.env, &type_inst);
+                emitln!(writer, "$ghost_{} := {};", var_name, var_name);
+            }
+        }
+
```

Comment:
I see what's going on, in the case of `self.style == FunctionTranslationStyle::Opaque`, the ghost variables are saved here for the call to `self.function_variant_name(FunctionTranslationStyle::Aborts)` later on. The shouldn't be saved here, they should be loaded just before the call.

Reply:
this actually works ok